geotags/views.py

```diff
@@ -47,16 +49,18 @@
             ORDER BY sounds_sound.created DESC
                      %(end)s"""
 
-def generate_json(sound_queryset):
-    # When using Django Sound queries:
-    #sounds_data = [[s.id, s.geotag.lat, s.geotag.lon] for s in sound_queryset]
+def generate_bytearray(sound_queryset):
+    # sounds as bytearray
+    packed_sounds = cStringIO.StringIO()
+    for sound in sound_queryset:
+        packed_sounds.write(struct.pack("i", sound.id))
+        packed_sounds.write(struct.pack("i", int(sound.lat*1000)))
+        packed_sounds.write(struct.pack("i", int(sound.lon*1000)))
 
-    sounds_data = [[s.id, s.lat, s.lon] for s in sound_queryset if not math.isnan(s.lat) and not math.isnan(s.lon)]
```

Comment:
Please make this check too in `generate_bytearray` version. It was added to avoid "corrupted" geotags to break the whole map. I know the problem is "corrupted" geotags but anyway, we don't want user-facing things to break.